dai_cadcad/policies.py

```diff
@@ -439,6 +440,41 @@ def flipper_deal(flipper, vat, cat, bid_id, now):
 # ---
 
 
+# Keeper
+
+
+def keeper_bid(user_id, model, flipper, vat):
+    """ Executes a `keeper_bid`.
+    """
+    bid_id = uuid4().hex
+
+    std_input = {"bid": 0}
+    model = models.choose[model]
+    bid = model(std_input)
+    flipper_tend(flipper, vat, bid_id, user_id, flipper.lot, bid, flipper.now)
+    return {}
+
+
+def keeper_bid_generator(_params, _substep, _state_hist, state):
+    """ Executes all `keeper_bid` policies for a timestep.
+    """
+
+    bid_buffer = state["flipper_eth"]["bids"]
+    # assuming state is {1: "simple", 2, "medium", 3: "complex"}
+    keepers = state["keepers"]
+    vat = state["vat"]
+    for auction in bid_buffer:
```

Comment:
Have this be a `for i in range(len(bid_buffer))`, `i` will actually be the `bid_id` you pass into `keeper_bid`.
You can grab the bid using `bid_buffer[i]`.